src/components/modal/z-modal/styles.css

```diff
@@ -46,6 +46,8 @@
 .modal-container > header {
   padding: calc(var(--space-unit) * 2 - 2px) calc(var(--space-unit) * 2);
   background: var(--color-surface02);
+  border-top-left-radius: var(--border-radius);
```

Comment:
Guarda com'è attualmente in produzione, questi border radius a mobile non mi sembra che ci siano.